src/core/geometry/qgslinestring.cpp

```diff
@@ -1093,12 +1093,6 @@ void QgsLineString::setPoints( size_t size, const double *x, const double *y, co
 {
   clearCache(); //set bounding box invalid
 
-  if ( size <= 0 )
```

Comment:
Couldn't this shortcut just be adapted to `if ( size == 0 )` ?

Reply:
yes! done